src/excel/Workbook.js

```diff
@@ -38,6 +38,18 @@ export class Workbook extends NamedItemContainer {
     return new Worksheet(this._oneDrive, `${this._uri}/worksheets`, name, this._log);
   }
 
+  async createSession() {
+    const uri = `${this.uri}/createSession`;
+    const result = await this._oneDrive.doFetch(uri, false, {
+      method: 'POST',
+    });
+    return result.id;
+  }
+
+  setSessionId(sessionId) {
+    this._oneDrive.setWorkbookSessionId(this._uri, sessionId);
+  }
```

Comment:
shouldn't this be implicitly set by `createSession()` ?

Reply:
agree, this looks wrong. there is also no guard against calling `createSession` when it already has a session, or overwriting an existing session.

Reply:
yeah, this should be set by `createSession`, but since createSession call is expensive (~3 seconds), we would not want to call `createSession` everytime.
The client is persisting the session id and is providing the sessionId with every subsequent request. `createSession` is only invoked when there is no sessionId (initial call).

Alternatively, we can do something like :
```
async getOrCreateSession(sessionId): Promise<string> {
   if (sessionId) {
       this._oneDrive.setWorkbookSessionId(this._uri, sessionId);
       return sessionId;
  } else {
         //create session call 
  }
}
```

Reply:
> But we need setSessionId to be exposed to the client.

Why?

Reply:
> Why?

once generated, the client already has the sessionId persisted. The client is making multiple http calls during the session duration to the service (that will not have any memory of sessionId generated in previous calls as the service reinitialises the OneDrive Object everytime). This method will enable the service to set the sessionId without creating a new session everytime.

Reply:
> (that will not have any memory of sessionId generated in previous calls as the service reinitialises the OneDrive Object everytime). This method will enable the service to set the sessionId without creating a new session everytime.

ah. ok. so it's more of a constructor option, than a runtime option